core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala

```diff
@@ -743,6 +745,52 @@ class ReplicaFetcherThreadTest {
     verify(mockBlockingSend)
   }
 
+  @Test
+  def shouldReassignmentBytesInMetricsGetsUpdated(): Unit = {
+    val props = TestUtils.createBrokerConfig(1, "localhost:1234")
+    val config = KafkaConfig.fromProps(props)
+
+    val mockBlockingSend: BlockingSend = createNiceMock(classOf[BlockingSend])
+
+    val log: Log = createNiceMock(classOf[Log])
+
+    val partition: Partition = createNiceMock(classOf[Partition])
+    expect(partition.localLogOrException).andReturn(log)
+    expect(partition.isReassigning).andReturn(true)
+    expect(partition.isAddingLocalReplica).andReturn(true)
+
+    val replicaManager: ReplicaManager = createNiceMock(classOf[ReplicaManager])
+    expect(replicaManager.nonOfflinePartition(anyObject[TopicPartition])).andReturn(Some(partition))
+    val brokerTopicStats = new BrokerTopicStats
+    expect(replicaManager.brokerTopicStats).andReturn(brokerTopicStats)
+    expect(replicaManager.brokerTopicStats).andReturn(brokerTopicStats)
+
+    val replicaQuota: ReplicaQuota = createNiceMock(classOf[ReplicaQuota])
+
+    val thread = new ReplicaFetcherThread(
+      name = "bob",
+      fetcherId = 0,
+      sourceBroker = brokerEndPoint,
+      brokerConfig = config,
+      failedPartitions = failedPartitions,
+      replicaMgr = replicaManager,
+      metrics =  new Metrics(),
+      time = new SystemTime(),
+      quota = replicaQuota,
+      leaderEndpointBlockingSend = Some(mockBlockingSend))
+
+    replay(mockBlockingSend, replicaManager, partition, log, replicaQuota)
+
+    val records = MemoryRecords.withRecords(CompressionType.NONE,
+      new SimpleRecord(1000, "foo".getBytes(StandardCharsets.UTF_8)))
+
+    val partitionData: thread.FetchData = new FetchResponse.PartitionData[Records](
+      Errors.NONE, 0, 0, 0, Optional.empty(), Collections.emptyList(), records)
+    thread.processPartitionData(t1p0, 0, partitionData)
+
+    assertTrue("No reassignment bytes in recorded", brokerTopicStats.allTopicsStats.reassignmentBytesInPerSec.get.count() > 0)
```

Comment:
Great job on these tests. Do you mind adding in an assertion for `brokerTopicStats.allTopicsStats.bytesInRate`?

Reply:
Didn't you mean `replicationBytesIn`? `bytesInRate` is recorded on produce.